src/dmd/backend/cod3.d

```diff
@@ -3707,7 +3707,15 @@ void prolog_loadparams(ref CodeBuilder cdb, tym_t tyf, bool pushalloc, regm_t* n
 
             type *t = s.Stype;
             type *t2 = null;
-            if (tybasic(t.Tty) == TYstruct && config.exe != EX_WIN64)
+
+            tym_t tyb = tybasic(t.Tty);
+
+            if (tybasic(t.Tty) == TYarray && t.Tdim == 1 && !(t.Tty & mTYvolatile))
```

Comment:
I don't fully comprehend this fix, but it appears to be addressing the symptom rather than the root cause.  I added a note to the bugzilla issue that this issue appears only on Linux, but not on Windows.  That makes the `&& config.exe != EX_WIN64` condition look suspicious.  Perhaps remove that and see?

Reply:
Also, if this is a regression, it would be helpful to know which PR introduced the issue to better understand the underlying cause.